.github/workflows/build.yml

```diff
@@ -0,0 +1,163 @@
+# SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+#
+# SPDX-License-Identifier: MIT
+
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false #true
+      matrix:
+        os:
+          - ubuntu-22.04
+          - windows-2022
+          - macos-12
+        build_type:
+          - Debug
+          #- Release
+        client_only:
+          - client_only
+          - client_and_ui
+        qt:
+          - version: "5.15.2"
+            requested: "5.15"
+          - version: "6.5.3"
+            requested: "6.5.*"
+            modules: qtshadertools
+
+    steps:
+      - name: Install Qt with options and default aqtversion
+        uses: jurplel/install-qt-action@v3
+        with:
+          aqtversion: null # use whatever the default is
+          modules: ${{ matrix.qt.modules }}
+          version: ${{ matrix.qt.requested }}
+          cache: true
+
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Create build directory
+        run: mkdir build
+
+      - name: Install ninja-build tool (must be after Qt due PATH changes)
+        uses: turtlesec-no/get-ninja@main
+
+      - name: Make sure MSVC is found when Ninja generator is in use
+        if: ${{ runner.os == 'Windows' }}
+        uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Configure project
+        run: >
+          cmake -S . -B ./build -G Ninja
+          -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
+          -DGAMMARAY_CLIENT_ONLY_BUILD=${{ matrix.client_only.client_only }}
```

Comment:
Oh ­— actually, this _doesn't_ work. All of the configured jobs have `-DGAMMARAY_CLIENT_ONLY_BUILD=` on the CMake command line, which is why your tests didn't fail due to the missing probes that it _would_ otherwise have caused.

(Also, Windows client only builds are currently broken due to some more bad target-creation logic I didn't catch (because it doesn't affect Linux) in `plugins/quickinspector/CMakeLists.txt`. I'll have a PR submitted in a couple of minutes to fix that.)

Reply:
right, I initially had true and false but was not readable on the github matrix, do forgot to update it